girder/api/v1/collection.py

```diff
@@ -48,15 +48,34 @@ def __init__(self):
     @filtermodel(model=CollectionModel)
     @autoDescribeRoute(
         Description('List or search for collections.')
+        .notes('If you pass a "query" field, you can also pass a "mode" field'
+               'to select a different search mode. By default a full-text search'
+               'is provided.')
         .responseClass('Collection', array=True)
-        .param('text', 'Pass this to perform a text search for collections.', required=False)
+        .param('query', 'Pass this to perform a search for collections.', required=False)
+        .param('mode', 'Pass to search with a different search mode.', required=False)
```

Comment:
I suggest you still support the "text" param and mark it as deprecated.

In the notes, you could add something like:

```
"""By default the search mode is "text" and specifying the "query" field performs a full text-search.
Setting the "mode" to something allows to change the search behavior. Available mode are %s.
""" % search._allowedSearchMode
```